src/libraries/Microsoft.PowerFx.Core/Utils/DName.cs

```diff
@@ -1,179 +1,205 @@
+        public DName(string value)
+        {
+            Contracts.Assert(IsValidDName(value));
+            _value = value;
+        }
+
+        /// <summary>
+        /// The value of the name.
+        /// </summary>
+        public string Value => _value ?? string.Empty;
+
+        /// <summary>
+        /// Whether the name is valid.
+        /// </summary>
+        public bool IsValid => _value != null;
+
+        /// <summary>
+        /// String representation of the name value.
+        /// </summary>
+        /// <param name="name"></param>
+        public static implicit operator string(DName name) => name.Value;
+
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            return Value;
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return Value.GetHashCode();
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            Contracts.AssertValueOrNull(obj);
+
+            if (!(obj is DName))
+            {
+                return false;
+            }
+
+            return Equals((DName)obj);
+        }
+
+        /// <summary>
+        /// Whether two names are equal.
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public bool Equals(DName other)
+        {
+            return Value == other.Value;
+        }
+
+        /// <summary>
+        /// Whether the name is equal to a string value.
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public bool Equals(string other)
+        {
+            Contracts.AssertValueOrNull(other);
+            return Value == other;
+        }
+
+        public static bool operator ==(DName name1, DName name2) => name1.Value == name2.Value;
+
+        public static bool operator ==(string str, DName name)
+        {
+            Contracts.AssertValueOrNull(str);
+            return str == name.Value;
+        }
+
+        public static bool operator ==(DName name, string str)
+        {
+            Contracts.AssertValueOrNull(str);
+            return name.Value == str;
+        }
+
+        public static bool operator !=(DName name1, DName name2) => name1.Value != name2.Value;
+
+        public static bool operator !=(string str, DName name)
+        {
+            Contracts.AssertValueOrNull(str);
+            return str != name.Value;
+        }
+
+        public static bool operator !=(DName name, string str)
+        {
+            Contracts.AssertValueOrNull(str);
+            return name.Value != str;
+        }
+
+        /// <summary>
+        /// Returns whether the given name is a valid <see cref="DName" />. 
+        /// </summary>
+        /// <param name="strName"></param>
+        /// <returns></returns>
+        public static bool IsValidDName(string strName)

```

Comment:
>IsValidDName [](http://example.com/codeflow?start=27&length=12)

Is there a potential regression here where a field was invalid but unused - so it used to be ignored but now is causing a failure?

Reply:
reverted since the previous check also checked for a lot of the whitespace characters already and it's safer to keep previous validation.